script-property-substitution/src/test/java/org/frankframework/extentions/script/EmbeddedScriptEvaluationTest.java

```diff
@@ -0,0 +1,96 @@
+package org.frankframework.extentions.script;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import org.frankframework.util.StringResolver;
+
+class EmbeddedScriptEvaluationTest {
+
+	@Test
+	void testResolveBasic() {
+		// Arrange
+		EmbeddedScriptEvaluation embeddedScriptEvaluation = new EmbeddedScriptEvaluation();
+
+		Map<String, Object> map = new HashMap<>();
+		map.put("key", "value");
+
+		// Act
+		Optional<String> value = embeddedScriptEvaluation.resolve("=key", map, null, null, StringResolver.DELIM_START, StringResolver.DELIM_STOP, false);
+
+		// Assert
+		assertTrue(value.isPresent());
+		assertEquals("value", value.get());
+	}
+
+	@Test
+	void testResolveSimpleCalculation() {
+		// Arrange
+		EmbeddedScriptEvaluation embeddedScriptEvaluation = new EmbeddedScriptEvaluation();
+
+		Map<String, Object> map = new HashMap<>();
+
+		// Act
+		Optional<String> value = embeddedScriptEvaluation.resolve("=4 * 5", map, null, null, StringResolver.DELIM_START, StringResolver.DELIM_STOP, false);
+
+		// Assert
+		assertTrue(value.isPresent());
+		assertEquals("20", value.get());
+	}
+
+	@Test
+	void testResolveWithTransform() {
+		// Arrange
+		EmbeddedScriptEvaluation embeddedScriptEvaluation = new EmbeddedScriptEvaluation();
+
+		Map<String, Object> map = new HashMap<>();
+		map.put("key", "value");
+
+		// Act
+		Optional<String> value = embeddedScriptEvaluation.resolve("=key.toUpperCase()", map, null, null, StringResolver.DELIM_START, StringResolver.DELIM_STOP, false);
```

Comment:
Deze is leuk, kan je ook `=${instance.name}.toUpperCase()` doen? en wat gebeurd er als hij niet bestaat?